docs/DeveloperGuide.md

```diff
@@ -1,31 +1,400 @@
+
+An ArrayList of `Question` objects stores all the selected questions meant for the quiz.
+
+#### 2.3.5 Design of Stats Feature
+
+E-Duke-8 allows for user’s stats to be shown to the user when requested. These stats correspond to the results of the user’s past attempts of the quiz. An aggregate result, followed by topical results of the quiz will be displayed. 
+
+A `Stats` class facilitates what is to be shown to the user. It also calls the methods of the two subclasses of `StatsCalculator` to retrieve the necessary information to be displayed.
+
+The class diagram given below showcases the high-level design of the stats feature in E-Duke-8. Given below it is a quick overview of each component.
+
+![Stats_Class_Diagram](./images/Stats.png)
+
+Results of the quiz attempts can be calculated using the information stored in a `Question` object, because of its methods, namely `wasShown()`,  `wasHintShown()` and `wasAnsweredCorrectly()`, that indicate if it has been attempted before, whether hint was used when user attempted the question and if the question was answered correctly respectively. 
+
+The current design of the stats feature is such that a correct answer without hint being used would award the user with 2 points, while a correct answer with hint used would award the user with 1 point. No point is awarded to the user if he chose the wrong answer. `calculatePointsEarnedForQuestion( :Question)’ is the method that contains the logic for this calculation. The method is explored further in the subsequent section. 
+
+An object of `UserStatsCalculator` class is responsible for calculating the aggregate results from the user’s previous quiz results. For instance, its `calculateTotalPointsEarned()` method will iterate through the multiple topics stored in E-Duke-8 and calculate the total sum of the user’s past results of the quizzes done for those topics.
+
+On the other hand, an object of `TopicStatsCalculator` is used by the object of `Stats` class to calculate the topical results. In its constructor, the `TopicStatsCalculator` object uses the single `Topic` object passed into it to retrieve its specific `QuestionList` object. Thereafter, by iterating through the questions for the particular `QuestionList` object, the results for individual topics can be calculated with its methods.
+
+
+
+### 2.4. Storage Component
+
+#### 2.4.1. Design of TopicsStorage
+
+Given data for the topics and questions is loaded automatically from JSON (JavaScript Object Notation) files in the data folder. This is mainly facilitated through the `TopicsStorage` 
+class which handles accessing the file as well as converting from JSON into `Topic`, `Question` and `Option` objects. The class diagram below shows this relationship.
+
+![TopicsStorage Class Diagram](./images/TopicsStorage.png)
```

Comment:
Could the arrows in the top right corner be combined? Just to make it less cluttered.
![image](https://user-images.githubusercontent.com/47075579/97395891-e8c7fb00-1920-11eb-8393-888dd23a651a.png)

Reply:
These lines are slightly confusing to look at, perhaps consider changing the placement of Topic, Question and Option(side by side so the arrows all go up and do not intersect each other).